src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -412,31 +409,90 @@ def output_schema(self) -> Type[BaseModel]:
         """
         return TextGenerationOutput
 
+    def _process_generation_config(
+        self, generation_config: [None, str, pathlib.Path, Dict, GenerationConfig]
+    ) -> GenerationConfig:
+        """
+        Process and return a GenerationConfig. The function can take in a filepath
+        pointing to a json consisting of the config values, a dictionary with the config
+        values, or a loaded GenerationConfig object. If None is given, the defaults are,
+        the pipeline GenerationConfig is used, if provided. If both are None, default
+        are used for generation.
+
+        :param generation_config: either a json filepath, dictionary or loaded
+        GenerationConfig object
+
+        :return: GenerationConfig object or None
+
+        """
+        if isinstance(generation_config, GenerationConfig):
+            return generation_config
+
+        if not generation_config:
+            return None
+
+        # TODO: move to tmp folder
+        if isinstance(generation_config, dict):
+            config_dir = os.getcwd()
+            config_name = "generation_config.json"
+            local_config_path = os.path.join(config_dir, config_name)
+            _LOGGER.info(
+                "Dictionary provided for the generation config. Creating temporary "
+                " generation_config.json"
+            )
+            with open(local_config_path, "w") as f:
+                json.dump(generation_config, f)
+
+        if isinstance(generation_config, (str, pathlib.Path)):
+            generation_config = pathlib.Path(generation_config)
+            config_dir = generation_config.parent.absolute()
+            config_name = generation_config.name
+
+        generation_config = GenerationConfig.from_pretrained(config_dir, config_name)
+        return generation_config
+
     def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         """
         Convert the input schema for the pipeline to the inputs for the engine.
 
         :param inputs: the input schema for the pipeline
         :return: the inputs for the engine
         """
+        generation_config = self._process_generation_config(inputs.generation_config)
+        if generation_config is None:
+            if self.generation_config:
+                generation_config = self.generation_config
+        else:
+            _LOGGER.info(
+                "Input generation config detected. This will override any"
+                " config provided during pipeline creation."
+            )
+
+        if not generation_config:
+            _LOGGER.info(
+                " No GenerationConfig detected. Using GenerationDefaults values"
+            )
+            generation_config = GenerationDefaults()
+
+        num_return_sequences = generation_config.num_return_sequences
```

Comment:
are we updating this value before it's accessed? otherwise, let's just refer to the live config